File: Problems/No.1221_Split_a_String_in_Balanced_Strings/split_a_string_in_balanved_strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        
        // parameter
        int balanced = 0;
        int balanced_strings = 0;

        // start computing
        for (auto c: s) {

            if (c == 'L') balanced++;
            else balanced--;

            if (balanced == 0) balanced_strings++;

        }

        return balanced_strings;

    }
};

File: Problems/No.1221_Split_a_String_in_Balanced_Strings/split_a_string_in_balanved_strings.py
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """

        # parameter
        balanced = 0
        balanced_strings = 0
    
        # start computing
        for c in s:
            if (c == 'L'):
                balanced += 1
            else:
                balanced -= 1
            if (balanced == 0):
                balanced_strings += 1
        
        return balanced_strings
